flags.md

```diff
@@ -0,0 +1,141 @@
+--cpu:i386                            # tags: x86
+
+--define:bit32                        # tags: arm32
+--define:bit32                        # tags: x86
+
+--gcc.path:/usr/bin                   # tags: arm64
+--gcc.exe:aarch64-linux-gnu-gcc       # tags: arm64
+--gcc.linkerexe:aarch64-linux-gnu-gcc # tags: arm64
+
+--passC:'-m32'                        # tags: x86, gcc
+--passL:'-m32'                        # tags: x86, gcc
+
+# Developer
+
+--embedsrc:on                         # tags: dev
+--define:DEV                          # tags: dev
+--listCmd                             # tags: dev
+
+--define:DEBUG                        # tags: debug
+--debugger:on                         # tags: debug
+--debuginfo                           # tags: debug
+--linedir:on                          # tags: debug
+
+--define:PROFILE                      # tags: memprofile
+--profiler:off                        # tags: memprofile
+--stackTrace:on                       # tags: memprofile
+--define:memProfiler                  # tags: memprofile
+
+--define:PROFILE                      # tags: profiler
+--profiler:on                         # tags: profiler
+--stackTrace:on                       # tags: profiler
+
+--debugger:native                     # tags: profilenative
+
+# Compilation
+
+--define:OPTIMIZED                    # tags: optimized
+
+--define:strip                        # tags: release
+--passC:-flto                         # tags: release, unix
+--passL:-flto                         # tags: release, unix
+
+# Compilation: full
+
+--define:ssl                          # tags: full
+
+# Compilation: mini/web
+
+--define:SAFE                         # tags: safe
+--define:MINI                         # tags: mini
+--define:WEB                          # tags: web
+
+--verbosity: 3                        # tags: web
+
+--define:NOASCIIDECODE                # tags: noasciidecode
+--define:NOCLIPBOARD                  # tags: noclipboard
+--define:NODIALOGS                    # tags: nodialogs
+--define:NOERRORLINES                 # tags: noerrorlines
+--define:NOGMP                        # tags: nogmp
+--define:NOPARSERS                    # tags: noparsers
+--define:NOSQLITE                     # tags: nosqlite
+--define:NOWEBVIEW                    # tags: nowebview
+
+# Compilation: defaults
+--verbosity:1                         # tags: default
+--hints:on                            # tags: default
+--hint:ProcessingStmt:off             # tags: default
+--hint:XCannotRaiseY:off              # tags: default
+--warning:GcUnsafe:off                # tags: default
+--warning:CastSizes:off               # tags: default
+--warning:ProveInit:off               # tags: default
+--warning:ProveField:off              # tags: default
+--warning:Uninit:off                  # tags: default
+--warning:BareExcept:off              # tags: default
+--threads:off                         # tags: default
+--skipUserCfg:on                      # tags: default
```

Comment:
`SkipUserCfg` is essential to avoid collision with our compilation flags and don't mess things up.

Question:
* Should we also use `--skipParentCfg`?

source: https://nim-lang.org/docs/nims.html

Reply:
@drkameleon